docs/cvl/builtin.md

```diff
@@ -4,7 +4,84 @@ Built-in Rules
 The Prover has some built-in general-purpose rules that can be verified on a
 contract out-of-the-box.
 
+## Deep Sanity
+
+Enable with:
+```cvl
+use builtin rule deepSanity;
+```
+
+Ensure `--multi_assert_check` is enabled (will throw an error otherwise!).
+
+One can configure the number of branching nodes that will be selected (it will always pick those nodes that dominate more) by setting `--settings -maxNumberOfReachChecksBasedOnDomination=N`, default `N=10`.
+
+### Background and motivation
+
+Sanity rules are one of our main methods to setup a new code base for verification. It serves two needs:
+
+- Checking that the Prover is able to solve through a path of the code, and that there are no obvious vacuities.
+- Checking that the Prover is able to find said path in a reasonable amount of time for both pre-processing and SMT phases.
+
+However sanity rules are limited. As they only require to find a single path, it is not guaranteed that a fast running time for the sanity rule means that the checked method is easy. In addition, it may be able to find a path that does not go through a vacuity that does exist deeper down in the code of the program.
+
+For example, consider:
+```solidity
+function foo() {
+  ...
+  for (uint i = 0 ; i < array.len; i++) {
+    ...
+  }
+  ...
+}
+```
+
+One trivial way to pass sanity here is we can find a model where `array.len=0`. In that case, our sanity check didn’t even visit the loop, regardless of our loop configuration. Any branching in the code is potentially hiding important code. 
```

Comment:
replace with
One trivial way to pass sanity here is to find a model where `array.len=0`. In that case, our sanity check doesn’t visit the loop, regardless of our loop configuration. Any branching in the code is potentially hiding important code.